src/Yasumi/Provider/Ukraine.php

```diff
@@ -58,11 +69,85 @@ public function initialize(): void
 
         // Add other holidays
         $this->calculateChristmasDay();
-        $this->calculateSecondInternationalWorkersDay();
         $this->calculateVictoryDay();
         $this->calculateConstitutionDay();
         $this->calculateIndependenceDay();
         $this->calculateDefenderOfUkraineDay();
+        $this->calculateCatholicChristmasDay();
+    }
+
+    /**
+     * Adds a holiday to the holidays providers (i.e. country/state) list of holidays.
+     *
+     * @param Holiday $holiday Holiday instance (representing a holiday) to be added to the internal list
+     *                         of holidays of this country.
+     * @param bool $postpone Holidays on a weekend will be postponed to the next monday.
+     * @param bool $addOnlyPostpone If $postpone is true add holidays on a weekend on the postponed day.
+     */
+    public function addHoliday(Holiday $holiday, bool $postpone = true, bool $addOnlyPostpone = false): void
+    {
+        if (!$postpone || !$this->isWeekendDay($holiday)) {
+            parent::addHoliday($holiday);
+            return;
+        }
+
+        // Special case: Holiday on a weekend and should be postpone to monday.
+
+        // Add original holiday.
+        if (!$addOnlyPostpone) {
+            parent::addHoliday($holiday);
+        }
+
+        // Create postponed holiday.
+        $postponed = new Holiday(
+            $holiday->shortName . 'Postponed',
+            $holiday->translations,
+            $holiday,
+            $holiday->displayLocale,
+            self::TYPE_POSTPONED
+        );
+
+        // Holidays on weekends will be postponed to monday.
+        do {
+            $postponed->modify('+1 days');
+        } while ($this->isWeekendDay($postponed));
+
+        // Create add holiday.
+        parent::addHoliday($postponed);
+    }
+
+    /**
+     * Returns the number of defined holidays (for the given country and the given year).
+     * In case a holiday is substituted (e.g. observed), the holiday is only counted once.
+     *
+     * @param bool $ignorePostponedHolidays Do not count postponed holidays.
+     *
+     * @return int number of holidays
+     */
+    public function count(bool $ignorePostponedHolidays = true): int
```

Comment:
What is the purpose of this function?

Reply:
Not needed anymore due to usage of the `SubstituteHoliday` class.